### DISCUSSION TITLE
Map.Iterate on a closed map is not throwing error.

### DISCUSSION THREAD

**Discussion by vakalapa** [cilium/ebpf#840] [Q&A]
Hi, 

I am using v0.9.0. I closed the map accidentally and tried iterating on the map to get all keys and values. 
Iterate.Err() returned nil. Is this expected ? 

The map type i used is BPF_MAP_TYPE_PERCPU_HASH.

Thanks

**Comment by anubhabMajumdar** [comment]
> Hi,
> 
> I am using v0.9.0. I closed the map accidentally and tried iterating on the map to get all keys and values. Iterate.Err() returned nil. Is this expected ?
> 
> The map type i used is BPF_MAP_TYPE_PERCPU_HASH.
> 
> Thanks

I expected `Iterate.Next` to throw error if map is closed as well.

**Reply by florianl** [reply]
I tried to reproduce the described issue with [a slightly modified version](https://gist.github.com/florianl/68fcadaa8438be5a2386d43e775a1974) of [`ExampleMap_perCPU()`](https://github.com/cilium/ebpf/blob/a9cf21df64c2962c130e2f1b12d467d2fba0996a/map_test.go#L1897-L1899)  by calling `arr.Close()` while iterating over the map.
After the call to `arr.Close()` the call of `arr.Next()` returns false and `arr.Err()` returns `bad file descriptor`. So I can't confirm the described issue.
Can you provide a minimal code example to reproduce the issue?
